moveit_core/kinematic_constraints/src/utils.cpp

```diff
@@ -281,263 +283,248 @@ moveit_msgs::msg::Constraints constructGoalConstraints(const std::string& link_n
-//       ROS_WARN_STREAM_NAMED(LOGNAME, "orientation constraint contains unknown entity '" << it->first << "'");
-//     }
-//   }
-//
-//   constraint.sensor_view_direction = moveit_msgs::VisibilityConstraint::SENSOR_X;
-//
-//   return true;
-// }
-//
-// static bool collectConstraints(XmlRpc::XmlRpcValue& params, moveit_msgs::Constraints& constraints)
-// {
-//   if (params.getType() != XmlRpc::XmlRpcValue::TypeArray)
-//   {
-//     ROS_ERROR_NAMED(LOGNAME, "expected constraints as array");
-//     return false;
-//   }
-//
-//   for (int i = 0; i < params.size(); ++i)  // NOLINT(modernize-loop-convert)
-//   {
-//     if (!params[i].hasMember("type"))
-//     {
-//       ROS_ERROR_NAMED(LOGNAME, "constraint parameter does not specify its type");
-//     }
-//     else if (params[i]["type"] == "joint")
-//     {
-//       constraints.joint_constraints.emplace_back();
-//       if (!constructConstraint(params[i], constraints.joint_constraints.back()))
-//         return false;
-//     }
-//     else if (params[i]["type"] == "position")
-//     {
-//       constraints.position_constraints.emplace_back();
-//       if (!constructConstraint(params[i], constraints.position_constraints.back()))
-//         return false;
-//     }
-//     else if (params[i]["type"] == "orientation")
-//     {
-//       constraints.orientation_constraints.emplace_back();
-//       if (!constructConstraint(params[i], constraints.orientation_constraints.back()))
-//         return false;
-//     }
-//     else if (params[i]["type"] == "visibility")
-//     {
-//       constraints.visibility_constraints.emplace_back();
-//       if (!constructConstraint(params[i], constraints.visibility_constraints.back()))
-//         return false;
-//     }
-//   }
-//
-//   return true;
-// }
-//
-// bool constructConstraints(XmlRpc::XmlRpcValue& params, moveit_msgs::Constraints& constraints)
-// {
-//   if (!isStruct(params, { "name", "constraints" }, "Parameter"))
-//     return false;
-//
-//   constraints.name = static_cast<std::string>(params["name"]);
-//   return collectConstraints(params["constraints"], constraints);
-// }
-// }  // namespace kinematic_constraints
+  std::vector<double> orientation;
+  if (!node->get_parameter(pose_param + "orientation", orientation) || orientation.size() != 3)
+    return false;
+
+  tf2::Quaternion q;
+  q.setRPY(orientation[0], orientation[1], orientation[2]);
+  pose.pose.orientation = toMsg(q);
+
+  std::vector<double> position;
+  if (!node->get_parameter(pose_param + "position", position) || position.size() != 3)
+    return false;
+
+  pose.pose.position.x = position[0];
+  pose.pose.position.y = position[1];
+  pose.pose.position.z = position[2];
+
+  return true;
+}
+
+static bool constructConstraint(const rclcpp::Node::SharedPtr& node, const std::string& constraint_param,
+                                moveit_msgs::msg::JointConstraint& constraint)
+{
+  node->get_parameter(constraint_param + "weight", constraint.weight);
```

Comment:
Originally, I had `constraint_param` include the `"."` but I realized this is confusing. I changed it so that the `.` is always included in the strings.